### DISCUSSION TITLE
数据量较大时（我的是8000W万条数据），Attu的索引会一直转圈圈

### DISCUSSION THREAD

**Discussion by LiJie20190102** [milvus-io/milvus#23551] [Q&A and General discussion] [answered]
不知道是不是和数据量有关系，我遇到的是数据量很大时会这样。
Attu版本v2.2.3
milvus版本v2.2.6
![image](https://user-images.githubusercontent.com/53458004/233115789-fe182b09-b4c3-4dec-86c0-c3c4e84dc8b9.png)

**Comment by xiaofan-luan** [comment]
load只是比较慢，需要一些时间。
一般load都是异步做的，如果minio大的话会快一些。正常一般需要几分钟吧

**Reply by LiJie20190102** [reply]
此时我还没进行load操作，只是将数据插入进去，我打开attu界面，就开始转圈圈。我想着要是attu的问题，我就可以忽略掉，我去使用java-sdk操作

**Reply by yhmo** [reply, marked as answer]
Building index for millions of vectors might take minutes.
Index time cost is affected by the following factors:
1. vector dimension
2. row count
3. index parameters
4. CPU cores of index node

**Reply by LiJie20190102** [reply]
如果我是用java-sdk我如何知道我的索引已经构建好了呢

**Reply by yhmo** [reply]
> 如果我是用java-sdk我如何知道我的索引已经构建好了呢

可以用 https://milvus.io/api-reference/java/v2.2.5/Index/getIndexBuildProgress().md 或者 https://milvus.io/api-reference/java/v2.2.5/Index/getIndexState().md 来获取当前索引构建的状态
getIndexBuildProgress() 返回有多少行数已经构建了索引
getIndexState() 返回一个枚举IndexState的状态值，可能是InProgress, Finished, Failed
